R/db2.R

```diff
@@ -322,23 +322,41 @@ report_data_import <- function(con, workdir, config) {
   ## TODO: patch this back in for the saved rds I think
   hash_orderly_yml <- hash_files(file.path(workdir, "orderly.yml"), FALSE)
 
+  ## there might be a better way to this.
+  ## At the moment we are repeatedly checking if a readme has been copied
+  ## maybe add a field to the run info saying that we've copied a README?
+  if (file_exists(file.path(workdir, "README.md"), check_case = FALSE)) {
+    if (file_has_canonical_case("README.md")) {
+      readme_name <- file_canonical_case("README.md")
```

Comment:
This line is never hit in the tests - can we simplify the logic here perhaps?

Reply:
What I've done here is wrong I think.
The canonical_case (via `file_split_base`) needs the full absolute path, so this will always fail to find a canonical form. If we pass in the full path e.g. `file_canonical_case(file.path(workdir, "README.md"))` then this breaks elsewhere since internally, Orderly uses relative paths.

I'll have a think about the best way forward.